### PR TITLE
Update plugin readme + fix examples

### PR DESCRIPTION
**Pull request by restrry**
- update readme
- examples use latest `saga` related packages
- fix real-world, cancellable-counter examples

### CODE REVIEW

File: examples/async/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "test": "cross-env NODE_ENV=test babel-node test/sagas.js | tap-spec"
   },
   "dependencies": {
-    "redux-saga": "^1.0.0-beta.0"
+    "redux-saga": "file:../../packages/core"
```
Comment (Andarist) [inline examples/async/package.json]: actually I would prefer not relaying on those packages being in a monorepo and stuff, with "real" dependencies we can in example easily upload those examples to things like codesandbox.io etc and it should download its copies of those deps

File: examples/real-world/.babelrc.js [hunk 2]
```diff
@@ -9,7 +9,6 @@ module.exports = {
       '@babel/env',
       {
         loose: true,
-        modules: false,
         forceAllTransforms: true,
```
Comment (Andarist) [inline examples/real-world/.babelrc.js]: why the change here? this demo uses webpack and it's better not to transpile modules when using it

File: packages/babel-plugin-redux-saga/README.md [hunk 3]
```diff
@@ -19,44 +21,43 @@ function* saga2(){
...
-        res[_SAGA_LOCATION] = {
-            fileName: "{{filename}}",
-            lineNumber: 2,
-            code: "foo(1, 2, 3)"
-        };
-        return res;
-    }();
+    yield Object.defineProperty(call(foo, 1, 2, 3), Symbol.for("@@redux-saga/LOCATION"), {
+        value: {
+            fileName: "{{pathToFile}}",
```
Comment (Andarist) [inline packages/babel-plugin-redux-saga/README.md]: in readme it would be more clear to use some dummy path here like, `../src/sagas/index.js`

File: examples/async/package.json (hunk 1 above)
Comment (restrry) [reply examples/async/package.json]: ok. I'll revert it

File: examples/real-world/.babelrc.js (hunk 2 above)
Comment (restrry) [reply examples/real-world/.babelrc.js]: `server.js` is not built with webpack. second option is use CJS for server.js

File: packages/babel-plugin-redux-saga/README.md (hunk 3 above)
Comment (restrry) [reply packages/babel-plugin-redux-saga/README.md]: ok

File: packages/babel-plugin-redux-saga/README.md [hunk 4]
```diff
@@ -19,44 +22,43 @@ function* saga2(){
...
+})
 ```
 
 ## Usage
 
 1. with babel
 ```js
-babel.transform(content, {
-    sourceMaps: true,
+// .babelrc
```
Comment (Andarist) [inline packages/babel-plugin-redux-saga/README.md]: imho would be good to use a valid JSON here

**Comment by Andarist** [comment]
> now there are 3 local dependencies:

I think lerna should update used versions across all packages when bumping a version, but we'll see.

> server.js and its dependencies are not handled with webpack.

This should be covered with [this](https://github.com/redux-saga/redux-saga/blob/30646f5f15b72ca23cbccba45427d83024376fc2/examples/real-world/.babelrc.js#L20), but I forgot to use env var for this script, we should define `start` script as:
```js
"start": "cross-env BABEL_ENV=cjs node --require @babel/register server.js"
```
and I think it should work.

OTOH not sure why this example has webpack config and if it's used anywhere 😟

**Comment by restrry** [comment]
1 suggested approach works, but  actually it does the same thing as `module: false`. isn't it?
2 webpack.config.js is used [here](https://github.com/redux-saga/redux-saga/blob/master/examples/real-world/server.js#L6)

**Comment by Andarist** [comment]
1. suggested approach is to use opt-in to cjs format, while `modules: false` is an opt-out. In this case it's kinda an equivalent of `modules: 'commonjs'`

I generally like using `modules: false` everywhere as a rule of thumb and opting into cjs on demand with env vars. Generally tooling which supports esm supports cjs too and when using cjs u end up with unoptimized builds and in most cases you are not even warned about it. cjs-only tools fail immediately on esm though and let you opt-in into cjs format easily, so the problem is easy to spot.
